packages/client/lib/sync/fetcher/blockfetcher.ts

```diff
@@ -1,60 +1,45 @@
+const level = require('level')
 import { Fetcher, FetcherOptions } from './fetcher'
 import { Block, BlockBodyBuffer } from '@ethereumjs/block'
-import { BN } from 'ethereumjs-util'
 import { Peer } from '../../net/peer'
 import { EthProtocolMethods } from '../../net/protocol'
-import { Chain } from '../../blockchain'
-
-export interface BlockFetcherOptions extends FetcherOptions {
-  /* Blockchain */
-  chain: Chain
-
-  /* Block number to start fetching from */
-  first: BN
-
-  /* How many blocks to fetch */
-  count: BN
-}
+import VM from '@ethereumjs/vm'
+import { DefaultStateManager } from '@ethereumjs/vm/dist/state'
+import { SecureTrie as Trie } from '@ethereumjs/trie'
 
 /**
  * Implements an eth/62 based block fetcher
  * @memberof module:sync/fetcher
  */
 export class BlockFetcher extends Fetcher {
-  protected chain: Chain
-  protected first: BN
-  protected count: BN
+  public vm: VM
 
   /**
    * Create new block fetcher
-   * @param {BlockFetcherOptions}
+   * @param {FetcherOptions}
    */
-  constructor(options: BlockFetcherOptions) {
+  constructor(options: FetcherOptions) {
     super(options)
 
-    this.chain = options.chain
-    this.maxPerRequest = options.maxPerRequest ?? 128
-    this.first = options.first
-    this.count = options.count
-  }
+    if (!this.config.vm) {
+      const db = level('./statedir')
```

Comment:
Definitely, this was one of my clean-up tasks. I just added this since I needed to finish and wanted to have a proof that this works at all.

Slight modified suggestion: I would store this in a parallel `statedata` directory with the `chain` (e.g. `ropsten`) as the base directory, this would need some modifications:

- `datadir` in `Config` should reflect the base directory (so without the `chaindata` or `statedata` part
- `Config.getSyncDataDirectory()` (used in `bin/cli.ts`) should be renamed to `getSystemDefaultDataDirectory()` (or something similar) and also return the (then system specific (somewhat)) base directory
- The other parts (`chaindata` or `lightchaindata`,`statedata`should be added on the blockchain and trie instantiations

Does this make sense? If so feel free to just pick up.

Reply:
So under `datadir` we would have a `chaindata` and `statedata` directory? It is currently configured that this `datadir` can only point to a single chain, so if you want to run both mainnet and ropsten then you'd need two `datadir`s.

Reply:
Ah, likely `datadir` should also be without the chain, right? 🤔  So just the upper most level base dir without any configuration selection.